components/backup/src/writer.rs

```diff
@@ -49,16 +48,16 @@ impl From<CfNameWrap> for CfName {
     }
 }
 
-struct Writer {
-    writer: RocksSstWriter,
+struct Writer<E: KvEngine> {
+    writer: <E as SstExt>::SstWriter,
```

Comment:
Can we rely on the `SstWriter` itself (instead of generic by a `KvEngine` but only use its `SstWriter` type)?

Reply:
How about `Writer<W: SstWriter + 'static>`?

Reply:
So what's the advantage of change it and introduce lifetime?

Reply:
Use W: SstWriter + 'static instead. PTAL